WD-Firmware/comrv/comrv.c

```diff
@@ -18,34 +18,765 @@
+         {
+            /* get the candidate entry */
+            pEntry = &stComrvCB.stOverlayCache[ucEvictCandidateList[ucEntryIndex]];
+            /* calc the source address */
+            pAddress = ((u08_t*)pEntry->pFixedEntryAddress + M_COMRV_GET_OVL_GROUP_SIZE_IN_BYTES(pEntry->unToken));
+            /* perform code copy */
+            comrvMemcpyHook(pEntry->pFixedEntryAddress, pAddress,
+                  stComrvCB.stOverlayCache[ucEvictCandidateList[ucEntryIndex]].pFixedEntryAddress - pAddress);
+
+            /* after code copy we need to align the entries structures */
+            for(ucIndex = ucEvictCandidateList[ucEntryIndex] ; ucIndex < ucEvictCandidateList[ucEntryIndex+1] ; ucIndex++)
+            {
+               pEntry = &stComrvCB.stOverlayCache[ucIndex + stComrvCB.stOverlayCache[ucIndex].unProperties.stFields.ucSizeInMinGroupSizeUnits];
+#ifdef D_COMRV_OVL_DATA_SUPPORT
+               /* an overlay data is present when handling defragmentation */
+               if (pEntry->unProperties.ucData)
+               {
+                  M_COMRV_EXIT_CRITICAL_SECTION();
+                  comrvNotificationHook(D_COMRV_OVL_DATA_DEFRAG_ERR, unToken.ucValue);
+               }
+#endif /* D_COMRV_OVL_DATA_SUPPORT */
+               stComrvCB.stOverlayCache[ucIndex].unProperties = pEntry->unProperties;
+               stComrvCB.stOverlayCache[ucIndex].unToken.uiValue = pEntry->unToken.uiValue;
+#ifdef D_COMRV_EVICTION_LRU
+               stComrvCB.stOverlayCache[pEntry->ucPrevIndex].ucNextIndex = ucIndex;
+               stComrvCB.stOverlayCache[pEntry->ucNextIndex].ucPrevIndex = ucIndex;
+#elif defined(D_COMRV_EVICTION_LFU)
+#elif defined(D_COMRV_EVICTION_MIX_LRU_LFU)
+#endif /* D_COMRV_EVICTION_LRU */
+            }
+            ucEntryIndex++;
+         }
+         ucNumOfEvictionCandidates++;
+      }
+      /* this means the end user locked all entries */
+      else if (ucNumOfEvictionCandidates == 0)
+      {
+         M_COMRV_EXIT_CRITICAL_SECTION();
+         comrvNotificationHook(D_COMRV_NO_AVAILABLE_ENTRY_ERR, unToken.uiValue);
+      }
+      ucIndex = ucEvictCandidateList[ucEntryIndex];
+      /* update the entry access */
+      comrvUpdateHeapEntryAccess(ucIndex);
+      /* update the heap entry with the new token */
+      stComrvCB.stOverlayCache[ucIndex].unToken.uiValue = unToken.uiValue;
+      /* update the heap entry properties with the group size */
+      stComrvCB.stOverlayCache[ucIndex].unProperties.stFields.ucSizeInMinGroupSizeUnits = usOverlayGroupSize;
+      /* if evicted size is larger than requested size we need to update the remaining tail*/
+      ucSizeOfEvictionCandidates = ucEvictCandidateList[ucNumOfEvictionCandidates] - usOverlayGroupSize;
+      if (ucSizeOfEvictionCandidates != 0)
+      {
+         ucEntryIndex += usOverlayGroupSize;
+         stComrvCB.stOverlayCache[ucEntryIndex].unProperties.stFields.ucSizeInMinGroupSizeUnits = ucSizeOfEvictionCandidates;
+#ifdef D_COMRV_EVICTION_LRU
```

Comment:
move this after load

Reply:
I think we concluded that moving this isn't correct

Reply:
we decided that it is not useful to move the code - we'll handle context switch